src/router/chainable_router.rs

```diff
@@ -0,0 +1,71 @@
+use hyper::method::Method;
+use middleware::Middleware;
+use nickel::Nickel;
+use router::http_router::{HttpRouter};
+
+pub struct ChainableRoute<'a, D: Sync + Send + 'static = ()> {
+    path: String,
```

Comment:
Yeah I'm trying to hack around my limitations in understanding lifetimes.  I'll try that and see if I can get it working with an &str.